dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/http/EndpointRequest.java

```diff
@@ -27,12 +29,20 @@
 import io.openliberty.security.oidcclientcore.discovery.OidcDiscoveryConstants;
 import io.openliberty.security.oidcclientcore.exceptions.OidcClientConfigurationException;
 import io.openliberty.security.oidcclientcore.exceptions.OidcDiscoveryException;
+import io.openliberty.security.oidcclientcore.storage.CookieBasedStorage;
+import io.openliberty.security.oidcclientcore.storage.SessionBasedStorage;
+import io.openliberty.security.oidcclientcore.storage.Storage;
 
 @Component(service = EndpointRequest.class, immediate = true, configurationPolicy = ConfigurationPolicy.IGNORE)
 public class EndpointRequest {
 
     public static final TraceComponent tc = Tr.register(EndpointRequest.class);
-
+    protected String state;
+    protected Storage storage;
+    protected enum StorageType {
+        COOKIE, SESSION
+    }
+    protected StorageType storageType;
```

Comment:
Since EndpointRequest service might be a single instance and the state, storage, and storage type depend on the client config and request, having these fields here will cause a concurrency problem.

Reply:
okay. moved the handling storage instantiation back to its original place